src/model/registerConsent.model.ts

```diff
@@ -112,17 +119,76 @@ export class RegisterConsentModel
   }
 
   async onVerifyConsent (): Promise<void> {
-    // not sure what functions to use or if they are ready
-    // for now we are just going do nothing here.
-    // todo: update transition to
-    // - verify consent
-    // - store consent
-    // - throw errors if there are errors in verifying the consent
-    /*
     const { consentsPostRequestAUTH, participantDFSPId } = this.data
     try {
+      const derivedChallenge = deriveChallenge(consentsPostRequestAUTH)
+      const encodedDerivedChallenge = encodeBase64String(derivedChallenge)
+
+      const decodedJsonString = decodeBase64String(consentsPostRequestAUTH.credential.payload.response.clientDataJSON)
+      const parsedClientData = JSON.parse(decodedJsonString)
+
+      const attestationExpectations: ExpectedAttestationResult = {
+        challenge: encodedDerivedChallenge,
+        // not sure what origin should be here
+        // decoding and copying the origin for now
+        origin: parsedClientData.origin,
+        factor: 'either'
+      }
+
+      const f2l = new Fido2Lib()
+      const clientAttestationResponse: AttestationResult = {
+        id: str2ab(consentsPostRequestAUTH.credential.payload.id),
+        rawId: str2ab(consentsPostRequestAUTH.credential.payload.rawId),
+        response: {
+          clientDataJSON: consentsPostRequestAUTH.credential.payload.response.clientDataJSON,
+          attestationObject: consentsPostRequestAUTH.credential.payload.response.attestationObject
+        }
+      }
+
+      await f2l.attestationResult(
+        clientAttestationResponse,
+        attestationExpectations
+      )
     } catch (error) {
-      this.logger.push({ error }).error('start -> requestIsValid')
+      this.logger.push({ error }).error('start -> consentVerified')
+
+      let mojaloopError
+      // if error is planned and is a MojaloopApiErrorCode we send back that code
+      if ((error as Errors.MojaloopApiErrorCode).code) {
+        mojaloopError = reformatError(error, this.logger)
+      } else {
+        // if error is not planned send back a generalized error
+        mojaloopError = reformatError(
+          Errors.MojaloopApiErrorCodes.TP_ACCOUNT_LINKING_ERROR,
+          this.logger
+        )
+      }
+
+      await this.thirdpartyRequests.putConsentsError(
+        consentsPostRequestAUTH.consentId,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        participantDFSPId
+      )
+
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  async onStoreConsent (): Promise<void> {
+    const { consentsPostRequestAUTH, participantDFSPId } = this.data
+    try {
+      await createAndStoreConsent(
+        consentsPostRequestAUTH.consentId,
+        // initiatorId is deprecated so just using empty string for now
+        '',
+        // todo: double check if this is supposed to be the DFSP's id
```

Comment:
Hmm I think so...

Reply:
honestly theres still much to do on the auth-service
config needs to be refactored
old fido2 code cleanup
database schema changes maybe...

just trying to get this over the line

Reply:
but yes I concur in that I believe it's the DFSP's id so I'll remove the comment